pennylane/devices/qutrit_mixed/initialize_state.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Functions to prepare a qutrit mixed state."""
+
+from typing import Iterable, Union
+import numpy as np
+import pennylane as qml
+from pennylane import (
+    QutritBasisState,
+)
+from pennylane.operation import (
+    StatePrepBase,
+    Operation,
+)
+from pennylane.wires import Wires
+
+qudit_dim = 3  # specifies qudit dimension
+
+
+def create_initial_state(
+    wires: Union[qml.wires.Wires, Iterable],
+    prep_operation: Operation = None,
+    like: str = None,
+):
+    r"""
+    Returns an initial state, defaulting to :math:`\ket{0}\bra{0}` if no state-prep operator is provided.
+
+    Args:
+        wires (Union[Wires, Iterable]): The wires to be present in the initial state
+        prep_operation (Optional[StatePrepBase]): An operation to prepare the initial state
+        like (Optional[str]): The machine learning interface used to create the initial state.
+            Defaults to None
+
+    Returns:
+        array: The initial state of a circuit
+    """
+    if isinstance(wires, Wires):
+        wires = wires.tolist()
+
+    num_wires = len(wires)
+
+    if not prep_operation:
+        rho = _create_basis_state(num_wires, 0)
+
+    elif isinstance(prep_operation, QutritBasisState):
+        rho = _apply_basis_state(prep_operation.parameters[0], wires)
+
+    elif isinstance(prep_operation, StatePrepBase):
+        rho = _apply_state_vector(prep_operation.state_vector(wire_order=list(wires)), num_wires)
+
```

Comment:
Is error handling done under the hood in operations like `prep_operation.state_vector`? Otherwise a user could pass in an integer outside the allowable range, or a non-normalized state vector. Worth looking into this, and adding appropriate tests (or throwing exceptions) as needed.

Reply:
Nope, I missed it as I was skimming the code. Thanks!

Reply:
Yes, such validation happens in either `prep_operation.__init__` or `prep_operation.state_vector`. You can see so in [QutritBasisState](https://github.com/PennyLaneAI/pennylane/blob/c7cda37e0b358394c9382dd0f8c30691e7f86da8/pennylane/ops/qutrit/state_preparation.py#L99), [BasisState](https://github.com/PennyLaneAI/pennylane/blob/c7cda37e0b358394c9382dd0f8c30691e7f86da8/pennylane/ops/qubit/state_preparation.py#L102), and [StatePrep](https://github.com/PennyLaneAI/pennylane/blob/c7cda37e0b358394c9382dd0f8c30691e7f86da8/pennylane/ops/qubit/state_preparation.py#L181).

Reply:
Also, error handling should also be done in preprocessing. 

[Docs on new device API:](https://docs.pennylane.ai/en/stable/code/api/pennylane.devices.Device.html)
If an arbitrary, non-preprocessed circuit is provided, [execute()](https://docs.pennylane.ai/en/stable/code/api/pennylane.devices.Device.html#pennylane.devices.Device.execute) has no responsibility to perform any validation or provide clearer error messages.